ecl/eclagent/eclagent.cpp

```diff
@@ -1864,6 +1864,7 @@ void EclAgent::doProcess()
         {
             MTIME_SECTION(queryActiveTimer(), "Process");
             Owned<IEclProcess> process = loadProcess();
+            QueryTerminationCleanup threadCleanup;
```

Comment:
The call to process() may install arbitrary hooks, if the ECL uses embedded languages (or unicode).